test/module/Entra/New-EntraInvitation.Tests.ps1

```diff
@@ -0,0 +1,230 @@
+                                                "UserType"                              = "Guest"
+                                            }
+                "InvitedUserDisplayName"  = ""
+                "InvitedUserEmailAddress" = "sanjeev.kumar@perennialsys.com"
+                "InvitedUserMessageInfo"  = @{
+                                                "CcRecipients"        = [System.Object]@()
+                                                "CustomizedMessageBody" = ""
+                                                "MessageLanguage"     = ""
+                                            }
+                "InvitedUserType"         = "Guest"
+                "ResetRedemption"         = $false
+                "SendInvitationMessage"   = $true
+                "Status"                  = "PendingAcceptance"
+                "AdditionalProperties"    = @{
+                                            "@odata.context" = "https://graph.microsoft.com/v1.0/$metadata#invitations/$entity"
+                }
+                "Parameters"              = $args
+            }
+        )
+    }
+    Mock -CommandName New-MgInvitation -MockWith $scriptblock -ModuleName Microsoft.Graph.Entra
+}
+
+Describe "New-EntraInvitation" {
+    Context "Test for New-EntraInvitation" {
+        It "Should invite a new external user to your directory" {
+            $result = New-EntraInvitation -InvitedUserEmailAddress sanjeev.kumar@perennialsys.com -SendInvitationMessage $True -InviteRedirectUrl "http://myapps.microsoft.com"
+            $result | Should -Not -BeNullOrEmpty
+            $result.Id | Should  -Be "11bb11bb-cc22-dd33-ee44-55ff55ff55ff"
+            $result.Status | Should -Be "PendingAcceptance"
+            $result.InvitedUser.Id | Should  -Be "bbbbbbbb-1111-2222-3333-cccccccccccc"
+            $result.InvitedUser.UserPrincipalName | Should  -Be "sanjeev.kumar@perennialsys.com"
+            $result.InvitedUser.UserType | Should  -Be "Guest"
+            $result.InvitedUserEmailAddress | Should -Be "sanjeev.kumar@perennialsys.com"
+            $result.InvitedUserType | Should -Be "Guest"
+            $result.ResetRedemption | Should -Be $false
+            $result.SendInvitationMessage | Should -Be $true
+            $result.InvitedUserDisplayName | Should -BeNullOrEmpty
+
+            Should -Invoke -CommandName New-MgInvitation -ModuleName Microsoft.Graph.Entra -Times 1
+        }
+
+        It "Should fail when parameters are empty" {
+            { New-EntraInvitation -InvitedUserEmailAddress -SendInvitationMessage -InviteRedirectUrl } | Should -Throw "Missing an argument for parameter*"
+        }
+
+        It "Should fail when InviteRedirectUrl parameter are Invalid" {
+            { New-EntraInvitation -InvitedUserEmailAddress sanjeev.kumar@perennialsys.com -SendInvitationMessage $True -InviteRedirectUrl "" } | Should -Throw "Cannot bind argument to parameter 'InviteRedirectUrl' because it is an empty string."
```

Comment:
```suggestion
            { New-EntraInvitation -InvitedUserEmailAddress SawyerM@contoso.com -SendInvitationMessage $True -InviteRedirectUrl "" } | Should -Throw "Cannot bind argument to parameter 'InviteRedirectUrl' because it is an empty string."
```